r/R/dataset-write.R

```diff
@@ -111,6 +121,10 @@ write_dataset <- function(dataset,
                           hive_style = TRUE,
                           existing_data_behavior = c("overwrite", "error", "delete_matching"),
                           max_partitions = 1024L,
+                          max_open_files = 900L,
+                          max_rows_per_file = 0L,
```

Comment:
It's a bit odd that the default `0L` here actually means no limit. In R I would expect either `NA`, `NULL`, or `Inf` to represent no limit.

For example, when using `write_parquet()` `chunk_size` defaults to `NULL`:

https://github.com/apache/arrow/blob/07ec0a12d430dc9151678b6f00d5c6fc0598f034/r/R/parquet.R#L144